torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -110,19 +119,32 @@ RegisterOperators reg({
         prim::Print,
         [](Node* node) {
           size_t num_inputs = node->inputs().size();
-          return [num_inputs](Stack& stack) {
+          std::string format = node->s(attr::string);
+          std::vector<std::string> strings;
+          for(size_t i = 0; i < format.size(); i++) {
+            if (format[i] == 's') {
+              strings.push_back(node->inputs()[i]->node()->s(attr::string));
+            }
+          }
+          return [num_inputs, format, strings](Stack & stack) {
+            auto inputs = last(stack, num_inputs);
             bool first = true;
-            for (at::Tensor i : last(stack, num_inputs)) {
-              if (!first)
-                std::cout << " ";
+            size_t str_i = 0;
+            for(size_t i = 0; i < format.size(); i++) {
+              if (!first) std::cout << " ";
               first = false;
-              if (auto tensor_impl = dynamic_cast<at::TensorImpl*>(i.get())) {
-                std::cout << at::Tensor(tensor_impl, true);
-              } else if (!i.defined()) {
-                std::cout << "<undefined tensor>";
+              if (format[i] == 's') {
+                std::cout << strings[str_i++];
```

Comment:
Use `.at` or assert that there are enough strings outside of this lambda (count `s`es, take the size of the vector). Otherwise we can end up with out of bounds accesses and that's a pain to debug